src/utils/fetchClient.ts

```diff
@@ -25,8 +27,8 @@ function request<T>(
       'Content-Type': 'application/json; charset=UTF-8',
     };
   }
-
   // for a demo purpose we emulate a delay to see if Loaders work
+
   return wait(300)
     .then(() => fetch(BASE_URL + url, options))
     .then(response => response.json());
```

Comment:
Consider adding error handling for the fetch request. If the response is not ok, you should throw an error or handle it appropriately to avoid unhandled promise rejections.